motionplan/plan.go

```diff
@@ -0,0 +1,154 @@
+package motionplan
+
+import (
+	"errors"
+	"fmt"
+	"math"
+
+	pb "go.viam.com/api/service/motion/v1"
+	"go.viam.com/rdk/motionplan/ik"
+	"go.viam.com/rdk/referenceframe"
+	"go.viam.com/rdk/spatialmath"
+)
+
+type Plan struct {
+	Trajectory
+	Path
+
+	// nodes corresponding to inputs can be cached with the Plan for easy conversion back into a form usable by RRT
+	// depending on how the Trajectory is constructed these may be nil and should be computed before usage
+	nodes []node
```

Comment:
In our abstractions, `nodes` are specifically nodes in an RRT graph. I know we only have RRT-based planners for now, but that could change. Baking a specific subclass of planner into our `Plan` object seems weird; perhaps `Plan` should be an interface?

Reply:
I've thought about making `Plan` an interface and ultimately leaned against it, the reason being articles I've read on premature interface creation as an antipattern.  We only have one type of Plan for now so it is a little overkill making it an interface, and future implementations of Plan might require different endpoints which will cause the interface definition to change 

https://medium.com/@cep21/preemptive-interface-anti-pattern-in-go-54c18ac0668a